### PR TITLE
Go to junit5, migrate tests

### PR DESCRIPTION
**Pull request by tomasalmeida**
- Migrate most of the tests to junit5
- Upgrade libs to junit 5 and mockito 4

### CODE REVIEW

**Comment by tomasalmeida** [comment]
Build failed due to

```
Error:  Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184:sonar (default-cli) on project accesscontroltool: Project not found. Please check the 'sonar.projectKey' and 'sonar.organization' properties, the 'SONAR_TOKEN' environment variable, or contact the project administrator -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184:sonar (default-cli) on project accesscontroltool: Project not found. Please check the 'sonar.projectKey' and 'sonar.organization' properties, the 'SONAR_TOKEN' environment variable, or contact the project administrator
```

could it be caused because I did a PR from the fork?

File: pom.xml [hunk 1]
```diff
@@ -123,27 +123,28 @@
...
                 <version>2.2</version>
                 <scope>test</scope>
             </dependency>
             <dependency>
-                <groupId>org.hamcrest</groupId>
-                <artifactId>hamcrest-library</artifactId>
-                <version>2.2</version>
+                <groupId>org.mockito</groupId>
+                <artifactId>mockito-core</artifactId>
+                <version>4.6.1</version>
```
Comment (kwin) [inline pom.xml:141]: should we introduce a property for the mockito version as I think both `mockito-core` and `mockito-junit-jupiter` are always released together (i.e. should have the same version)

**Review by kwin — APPROVED** [review]
LGTM, thanks a lot. Just one minor nitpick.

File: pom.xml (hunk 1 above)
Comment (tomasalmeida) [reply pom.xml:141]: I like it, I did not do that in the first place as the properties looked too specific and not version related, but yes, I do that tonight :-)

**Comment by tomasalmeida** [comment]
found an error on the migration, I am mixing it :-)

**Pull request closed without merging**